packages/overlays/docs/OverlayController.md

```diff
@@ -0,0 +1,192 @@
+
+- **contentNode**, the toggleable content of the overlay
+- **invokerNode**, the element that can trigger showing (and hiding) the overlay. For local overlays, this is the relative element the content is positioned to.
+
+For DOM position, local refers to overlays where the content is positioned next to the invokers they are related to, DOM-wise.
+Global refers to overlays where the content is positioned in a global root node at the bottom of `<body>`.
+
+## Configuration options
+
+In total, we should end up with configuration options as depicted below, for all possible overlays.
+All boolean flags default to 'false'.
+
+```text
+- {Boolean} trapsKeyboardFocus - rotates tab.
+- {Boolean} hidesOnEsc - hides the overlay when pressing [esc].
+- {Element} elementToFocusAfterHide - the element that should be called `.focus()` on after dialog closes.
+- {Boolean} hasBackdrop - whether it should have a backdrop. (local mode only)
+- {Boolean} isBlocking - hides other overlays when multiple are opened.
+- {Boolean} preventsScroll - prevents scrolling body content when overlay opened.
+- {Object} viewportConfig - placementMode: local only
+  - {String} placement: 'top-left' | 'top' | 'top-right' | 'right' | 'bottom-left' |'bottom' | 'bottom-right' | 'left' | 'center'
+- {Object} popperConfig - placementMode: local only
+  - {String} placement: 'top-left' | 'top' | 'top-right' | 'right' | 'bottom-left' |'bottom' | 'bottom-right' | 'left' | 'center'
+```
+
+> Note: popperConfig reflects [Popper.js API](https://popper.js.org/popper-documentation.html)
+
+## Specific Controllers
+
+You can find our existing configurations [here](../src/configurations):
+
+- withModalDialogConfig
+- withDropdownConfig
+- withBottomSheetConfig
+
+You import these using ES Modules, and then simply call them inside your OverlayController instantiation:
+
+```js
+const ctrl = new OverlayController({
+  ...withModalDialogConfig(),
+  invokerNode,
+  contentNode,
+});
+```
+
+## Responsive switching
+
+Currently we support switching between overlay configurations. Keep in mind however:
+
+- We do not yet support switching between overlay configurations while the content is shown. If you try, it will close the content if it is open, and the user will need to re-open. WIP.
+- We DO support switching the positioning configurations (`popperConfig` property for `local` mode, `viewportConfig` for `global` mode) while content is shown.
+
+An example implementation on an `OverlayController` instance which checks the viewport width, and then updates the configuration to a bottom sheet versus a modal dialog on `before-show`.
+
+```js
+myOverlayCtrl.addEventListener('before-show', () => {
+  if (window.innerWidth >= 600) {
+    ctrl.updateConfig(withModalDialogConfig());
+  } else {
+    ctrl.updateConfig(withBottomSheetConfig());
+  }
+});
+```
+
+An example implementation inside of a webcomponent that uses the `OverlayMixin`:
+Overriding protected method `_defineOverlay`.
```

Comment:
Ahh I see what you mean. But yeah, they are two different use cases: 
- inside a WC that implements OverlayMixin 
- outside without OverlayMixin but only with a OverlayController instance

They are quite important to separate, because sometimes people will extend a component or create one with the mixin, and sometimes people just want to use an existing component directly and add the switching behavior on it imperatively by targeting the existing component's Overlay Controller instance.

Reply:
I now see the reasoning behind those two examples. The point that you're trying to get across is that the OverlayController configuration can be updated at runtime? The OverlayMixin example illustrates that having an instance of the OverlayController can be used to do this. The first example does the same thing.

Reply:
Yes the event listener that updates the config can be added to the controller instance at runtime.

The second example is where you add the event listener as part of _defineOverlay (so on initialization, before run-time) method that is part of OverlayMixin (only works inside a Web Component that implements OverlayMixin).

So it's quite important to showcase both. But maybe you can suggest how I could make more clear in the docs why they are different.